### PR TITLE
[AST] Mark the fallthrough coreturn statement implicit.

### PR DESCRIPTION
**Pull request by hokein**

### CODE REVIEW

File: clang/lib/Sema/SemaCoroutine.cpp [hunk 1]
```diff
@@ -1743,7 +1743,7 @@ bool CoroutineStmtBuilder::makeOnFallthrough() {
       return false;
   } else if (HasRVoid) {
     Fallthrough = S.BuildCoreturnStmt(FD.getLocation(), nullptr,
-                                      /*IsImplicit*/false);
+                                      /*IsImplicit*/true);
```
Comment (cor3ntin) [inline clang/lib/Sema/SemaCoroutine.cpp:1746]:
```suggestion
                                      /*IsImplicit=*/true);
```

**Review by cor3ntin — APPROVED** [review]
LGTM modulo nit

**Comment by ChuanqiXu9** [comment]
Can you provide more context for this?

**Comment by ilya-biryukov** [comment]
> Can you provide more context for this?

I believe this is a follow-up to #77311.

**Review by ilya-biryukov — APPROVED** [review]
Approve from my side too, but please make sure to apply suggestion from @cor3ntin and clang-format before submitting.

**Review by ChuanqiXu9 — APPROVED** [review]
LGTM then.

**Pull request merged**